domain/application/state/application.go

```diff
@@ -173,7 +190,9 @@ func (st *State) CreateApplication(ctx domain.AtomicContext, name string, app ap
 				return errors.Annotatef(err, "creating channel row for application %q", name)
 			}
 		}
-		return nil
+
+		// insert application resources
+		return insertApplicationResources(ctx, tx)
```

Comment:
> I chose to not break things, so for now, nothing will be added in the resources table when creating an application.
> However, the state layer will log a warning if some resources declared in the charm are not added by create application. It could have been an error, though, since it is an inconsistent state.

^ from the PR description. It looks like you are just adding them here though.

Reply:
No because to trigger that thing we need to pass AddResourcesArgs to CreateApplication and we don't